sympy/functions/special/error_functions.py

```diff
@@ -1908,6 +1908,19 @@ def _eval_rewrite_as_Integral(self, z, **kwargs):
 
     _eval_rewrite_as_sinc =  _eval_rewrite_as_Integral
 
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):
+        arg = self.args[0].as_leading_term(x, logx=logx, cdir=cdir)
+        arg0 = arg.subs(x, 0)
+
+        if arg0 is S.NaN:
```

Comment:
if i understand correctly:

1. first you are checking that we don't have nan
2. then you menage the three caes: 0, finite but not zero and infinite

perhaps a small comment to make this clear above `if arg0.is_zero:` ;-)

Reply:
Not really required I would say. There are quite some _eval_as_leading_term implementations in SymPy (quite some of them like trig, inverse trig, hyperbolic have been done by me :) ) , so as this point I would say the template is pretty clear to us.